homeassistant/components/mqtt/mixins.py

```diff
@@ -314,10 +314,19 @@ async def _async_discover(
         raise
 
 
+class _SetupNonEntityHelperCallbackProtocol(Protocol):
+    """Callback protocol for async_setup in async_setup_non_entity_entry_helper."""
+
+    async def __call__(
+        self, config: ConfigType, discovery_data: DiscoveryInfoType
+    ) -> None:
+        ...
+
+
```

Comment:
How can it be this is not hit by the tests?

Reply:
It's only used for typing. I can add a comment to skip coverage if you like.

Reply:
That would be great

Reply:
Seem we can use:

`if TYPE_CHECKING:`

too